src/sys/termios.rs

```diff
@@ -1,648 +1,782 @@
+                      target_os = "openbsd"))]
+            BaudRate::B28800 => B28800,
+            BaudRate::B38400 => B38400,
+            BaudRate::B57600 => B57600,
+            #[cfg(any(target_os = "dragonfly",
+                      target_os = "freebsd",
+                      target_os = "macos",
+                      target_os = "netbsd",
+                      target_os = "openbsd"))]
+            BaudRate::B76800 => B76800,
+            BaudRate::B115200 => B115200,
+            BaudRate::B230400 => B230400,
+            #[cfg(any(target_os = "android",
+                      target_os = "freebsd",
+                      target_os = "linux",
+                      taget_os = "netbsd"))]
+            BaudRate::B460800 => B460800,
+            #[cfg(any(target_os = "android", target_os = "linux"))]
+            BaudRate::B500000 => B500000,
+            #[cfg(any(target_os = "android", target_os = "linux"))]
+            BaudRate::B576000 => B576000,
+            #[cfg(any(target_os = "android",
+                      target_os = "freebsd",
+                      target_os = "linux",
+                      taget_os = "netbsd"))]
+            BaudRate::B921600 => B921600,
+            #[cfg(any(target_os = "android", target_os = "linux"))]
+            BaudRate::B1000000 => B1000000,
+            #[cfg(any(target_os = "android", target_os = "linux"))]
+            BaudRate::B1152000 => B1152000,
+            #[cfg(any(target_os = "android", target_os = "linux"))]
+            BaudRate::B1500000 => B1500000,
+            #[cfg(any(target_os = "android", target_os = "linux"))]
+            BaudRate::B2000000 => B2000000,
+            #[cfg(any(target_os = "android", target_os = "linux"))]
+            BaudRate::B2500000 => B2500000,
+            #[cfg(any(target_os = "android", target_os = "linux"))]
+            BaudRate::B3000000 => B3000000,
+            #[cfg(any(target_os = "android", target_os = "linux"))]
+            BaudRate::B3500000 => B3500000,
+            #[cfg(any(target_os = "android", target_os = "linux"))]
+            BaudRate::B4000000 => B4000000,
         }
     }
+}
 
-    #[cfg(any(target_os = "linux", target_os = "android"))]
-    pub mod consts {
-        use libc::{c_int, c_uint, c_uchar};
-
-        pub type tcflag_t = c_uint;
-        pub type cc_t = c_uchar;
-        pub type speed_t = c_uint;
-
-        #[repr(C)]
-        #[derive(Clone, Copy)]
-        pub struct Termios {
-            pub c_iflag: InputFlags,
-            pub c_oflag: OutputFlags,
-            pub c_cflag: ControlFlags,
-            pub c_lflag: LocalFlags,
-            pub c_line: cc_t,
-            pub c_cc: [cc_t; NCCS],
-            pub c_ispeed: speed_t,
-            pub c_ospeed: speed_t
-        }
+/// Specify when a port configuration change should occur.
+///
+/// Used as an argument to `tcsetattr()`
+// TODO: Add TCSASOFT, which will require treating this as a bitfield.
+#[derive(Clone, Copy, Debug)]
+pub enum SetArg {
```

Comment:
In C these constants are always treated as `int`s, not `ssize_t`s.  You should add a `#[repr(i32)]` here, and ditch the `as isize` for each entry.